crates/matrix-sdk/src/room/timeline/event_handler.rs

```diff
@@ -698,11 +698,27 @@ impl<'a> TimelineEventHandler<'a> {
                 });
 
                 if let Some((idx, old_item)) = result {
-                    if old_item.as_remote().is_some() {
-                        // Item was previously received from the server. This
-                        // should be very rare normally, but with the sliding-
-                        // sync proxy, it is actually very common.
-                        trace!(?item, ?old_item, "Received duplicate event");
+                    if let Some(old_remote_item) = old_item.as_remote() {
+                        if matches!(
+                            old_remote_item.origin,
+                            RemoteEventOrigin::Cache | RemoteEventOrigin::Sync
+                        ) {
+                            // Item was previously received from syncing. This
+                            // should be very rare normally, but with the
+                            // sliding-sync proxy, it is actually very common.
+                            //
+                            // To avoid items being reordered only to eventually
+                            // end up in the same position after processing the
+                            // whole chunk, do nothing for this event.
+                            trace!(?item, ?old_item, "Received duplicate event, skipping");
+                            return;
+                        } else {
```

Comment:
nit: else after return (one can remove the `else` branch in that case, as it's part of the code just running after the `}`)